src/Uno.UI/UI/Xaml/UIElement.Pointers.Skia.cs

```diff
@@ -60,186 +107,402 @@ private void CoreWindow_PointerExited(CoreWindow sender, PointerEventArgs args)
-				var pointerArgs = new PointerRoutedEventArgs(args, element);
+#if TRACE_HIT_TESTING
+				using var _ = SET_TRACE_SUBJECT(element);
+				TRACE($"- hit test visibility: {elementHitTestVisibility}");
+#endif
 
-				if (element.RenderTransform != null)
+				// If the element is not hit testable, do not even try to validate it nor its children.
+				if (elementHitTestVisibility == HitTestVisibility.Collapsed)
 				{
-					currentTransform *= element.RenderTransform.MatrixCore;
-					elementRect = currentTransform.Transform(elementRect);
+					// Even if collapsed, if the element is stale, we search down for the real stale leaf
+					if (isStale?.Invoke(element) ?? false)
+					{
+						stale = SearchDownForStaleBranch(element, isStale);
+					}
+
+					TRACE($"> NOT FOUND (Element is HitTestVisibility.Collapsed) | stale branch: {stale?.ToString() ?? "-- none --"}");
+					return (default, stale);
 				}
 
-				if (elementRect.Contains(position))
+				// The region where the element was arrange by its parent.
+				// This is expressed in parent coordinate space
+				var layoutSlot = element.LayoutSlotWithMarginsAndAlignments;
+
+				// The maximum region where the current element and its children might draw themselves
+				// TODO: Get the real clipping rect! For now we assume no clipping.
+				// This is expressed in element coordinate space.
+				var clippingBounds = Rect.Infinite;
+
+				// The region where the current element draws itself.
+				// Be aware that children might be out of this rendering bounds if no clipping defined. TODO: .Intersect(clippingBounds)
+				// This is expressed in element coordinate space.
+				var renderingBounds = new Rect(new Point(), layoutSlot.Size);
+
+				// First compute the 'position' in the current element coordinate space
+				var posRelToElement = posRelToParent;
+
+				posRelToElement.X -= layoutSlot.X;
+				posRelToElement.Y -= layoutSlot.Y;
+
+				var renderTransform = element.RenderTransform;
+				if (renderTransform != null)
+				{
+					Matrix3x2.Invert(renderTransform.MatrixCore, out var parentToElement);
+
+					TRACE($"- renderTransform: [{parentToElement.M11:F2},{parentToElement.M12:F2} / {parentToElement.M21:F2},{parentToElement.M22:F2} / {parentToElement.M31:F2},{parentToElement.M32:F2}]");
+
+					posRelToElement = parentToElement.Transform(posRelToElement);
+					renderingBounds = parentToElement.Transform(renderingBounds);
+				}
+
+				if (element is ScrollViewer sv)
+				{
+					var zoom = sv.ZoomFactor;
+
+					TRACE($"- scroller: x={sv.HorizontalOffset} | y={sv.VerticalOffset} | zoom={zoom}");
+
+					posRelToElement.X /= zoom;
+					posRelToElement.Y /= zoom;
+
+					// No needs to adjust the position:
+					// On Skia the scrolling is achieved using a RenderTransform on the content of the ScrollContentPresenter,
+					// so it will already be taken in consideration by the case above.
+					//posRelToElement.X += sv.HorizontalOffset;
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6213625)